File: katas/sierpinski/spec/pascal_spec.js
var Pascal=require('../lib/pascal.js');

describe('genereTriangleDePascal', function () {
  it('doit renvoyer 1 pour un triangle à une ligne', function () {
    expect(Pascal.genereTriangle(1)[0]).toEqual([1]);
  });

  it('doit renvoyer 1,1 pour un triangle à 2 lignes', function () {
    expect(Pascal.genereTriangle(2)[1]).toEqual([1,1]);
  });

  it('doit renvoyer 1,2,1 pour un triangle à 3 lignes', function () {
    expect(Pascal.genereTriangle(3)[2]).toEqual([1,2,1]);
  });
});

describe('dessineTriangleDeSierpinski', function () {
  it("affiche * pour un triangle d'une ligne", function () {
    expect(Pascal.dessineTriangleDeSierpinski(1)).toEqual(['*']); 
  });

  it("affiche * * pour la 3e ligne du triangle", function () {
    expect(Pascal.dessineTriangleDeSierpinski(3)[2]).toEqual('* *'); 
  });

  it("affiche **** pour la 4e ligne du triangle", function () {
    expect(Pascal.dessineTriangleDeSierpinski(4)[3]).toEqual('****'); 
  });
});

File: katas/sierpinski/spec/pascal_spec.rb
require 'pascal'

describe Pascal do
  it "doit renvoyer comme première ligne 1" do
    expect(Pascal::triangle(1)).to eq([[1]])
  end

  it "doit renvoyer comme deuxieme ligne 1,1" do
    expect(Pascal::triangle(2)).to eq([[1],[1,1]])
  end

  it "doit renvoyer comme troisieme ligne 1,2,1" do
    expect(Pascal::triangle(3)).to eq([[1],[1,1],[1,2,1]])
  end

  it "doit renvoyer comme quatrieme ligne 1,3,3,1" do
    expect(Pascal::triangle(4)).to eq([[1],[1,1],[1,2,1],[1,3,3,1]])
  end

  it "doit renvoyer comme 300e ligne 300 elements" do
    expect(Pascal::triangle(300)[299].size).to eq(300)
  end

end
